---
output: html_document
---
```{r}
# loading all the required packages

rm(list=ls())

library(seqinr)
library(plyr)
library(dplyr)
library(car)
library(cowplot)
library(ggplot2)
library(Rsamtools)
library(sfsmisc)
library(plotROC)
library(mygene)

# REading depth files created from bedtools using output of samtools and Bowtie2 (please see scripts.sh file)
targetsData<-read.table("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/targetsHMVYVBCXX_1_GGCTAC_1.out", header=FALSE)
nontargetsData<-read.table("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/nontargetsHMVYVBCXX_1_GGCTAC_1.out", header=FALSE)
complementData<-read.table("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/complementHMVYVBCXX_1_GGCTAC_1.out", header=FALSE)
names(targetsData)<-c("Chromosome","Start","Stop", "Strand", "Gene", "Reads", "BasesCovered", "Length", "FractionCoverage")
names(nontargetsData)<-c("Chromosome","Start","Stop", "Strand", "Gene", "Reads", "BasesCovered", "Length", "FractionCoverage")
names(complementData)<-c("Chromosome","Start","Stop", "Reads", "BasesCovered", "Length", "FractionCoverage")


RL<-50 # Read length, if not using the individual depth, very comparable and complementary to using single position depth, however read length should be carefully used.

ATGC_content<-read.table("/Users/Vish/Downloads/Lorenzo/paper/targetsCoverageATGC.txt", header = TRUE)

non_ATGC_content<-read.table("/Users/Vish/Downloads/Lorenzo/paper/nontargetsCoverageATGC.txt", header = TRUE)

comp_ATGC_content<-read.table("/Users/Vish/Downloads/Lorenzo/paper/complementCoverageATGC.txt", header = TRUE)
    
targetsAllData<-merge(targetsData, ATGC_content, by=intersect(names(targetsData), names(ATGC_content)))

nontargetsAllData<-merge(nontargetsData, non_ATGC_content, by=intersect(names(nontargetsData), names(non_ATGC_content)))

complementAllData<-merge(complementData, comp_ATGC_content, by=intersect(names(complementData), names(comp_ATGC_content)))
    
targetsAllData$Region="Target"
nontargetsAllData$Region="Non-Target"
complementAllData$Region="Intergenic"

complementAllData$Strand<-'NA'
complementAllData$Gene<-'NA'

test<-rbind(targetsAllData,nontargetsAllData)
allData<-rbind(test,complementAllData)
allData$Depth<-allData$Reads/allData$Length

levels(allData$Region)<-levels(allData$Region)[3:1]
allData$Region <- factor(allData$Region, levels = c("Target", "Non-Target", "Intergenic"))

# only targets, non-targes and intergenic regions
filteredData <- subset(allData, Length >= 400, select=c(Depth, Region, Length, FractionCoverage))
filteredData$Depth<-filteredData$Depth*RL

# Only targets and non-targets
filteredDataTNT <- subset(allData, Length >= 400 & ( Region == "Target" | Region == "Non-Target"), select=c(Depth, Region, Length))
filteredDataTNT$D<-NULL
filteredDataTNT$D[which(filteredDataTNT$Region=="Target")]<-1
filteredDataTNT$D[which(filteredDataTNT$Region=="Non-Target")]<-0
filteredDataTNT$M<-filteredDataTNT$Depth

#pdf(file="//Users/Vish/Downloads/Lorenzo/pool/FASTQ/ROC_curve.pdf", width = 15, height = 15)
ROCplot <- ggplot(filteredDataTNT, aes(d = D, m = M)) + geom_roc(n.cuts = 0, color = "blue")+
  labs(x="False Positive Rate", y = "True Positive Rate")+#+style_roc()
theme(axis.title.y = element_text(face="bold", colour="black", size=20),axis.text.y  = element_text(vjust=0.5, size=20),
axis.title.x =element_text(face="bold", colour="black", size=20),axis.text.x  = element_text(vjust=0.5, size=20))+
  scale_y_continuous(expand=c(0,0),breaks=c(0,.25,.5,.75,1),labels=c(0,.25,.5,.75,1))+
  scale_x_continuous(expand=c(0,0),breaks=c(.25,.5,.75,1),labels=c(.25,.5,.75,1))
ROCplot<-ROCplot+annotate("text", x = .6, y = .3, size=9, color="blue",
           label = paste("AUC =", round(calc_auc(ROCplot)$AUC, 3))) 
ROCplot
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/ROCplot.pdf")
#dev.off()  

###################################################

adjust<-length(filteredData$FractionCoverage[which(filteredData$Region=="Target")])/length(filteredData$FractionCoverage[which(filteredData$Region=="Non-Target")])

adjustIG<-length(filteredData$FractionCoverage[which(filteredData$Region=="Target")])/length(filteredData$FractionCoverage[which(filteredData$Region=="Intergenic")])

stepsNontarget<-seq(adjust, length(filteredData$FractionCoverage[which(filteredData$Region=="Target")]), by=adjust)

stepsIntergenic<-seq(adjustIG, length(filteredData$FractionCoverage[which(filteredData$Region=="Target")]), by=adjustIG)

stepsTarget<-seq(1,length(filteredData$FractionCoverage[which(filteredData$Region=="Target")]), by=1)

filteredData<-filteredData[order(-filteredData$FractionCoverage), ]
filteredData$steps<-0
filteredData$steps[which(filteredData$Region=="Target")]<-stepsTarget
filteredData$steps[which(filteredData$Region=="Non-Target")]<-stepsNontarget
filteredData$steps[which(filteredData$Region=="Intergenic")]<-stepsIntergenic
filteredData$steps<-filteredData$steps/length(stepsTarget)

  
#pdf(file="//Users/Vish/Downloads/Lorenzo/pool/FASTQ/coverageT_NT_400nt_updated.pdf", width = 15, height = 15)
#plot(stepsTarget,sort(filteredData$FractionCoverage[which(filteredData$Region=="Target")], decreasing = TRUE),  col = "violet", xlim=c(1,4800), ylim = c(0,1),  xlab = "Index (arbitrary units)", ylab = "Fraction of sequence covered",lwd=0.5,cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5, yaxt="n", pch=19)
#par(new=TRUE)
#plot(stepsNontarget,sort(filteredData$FractionCoverage[which(filteredData$Region=="Non-Target")], decreasing = TRUE),  col = "green", xlim=c(1,4800), ylim = c(0,1), xlab = "Index (arbitrary units)", ylab = "Fraction of sequence covered",lwd=0.5,cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5, yaxt="n", pch=19)
#par(new=TRUE)
#plot(stepsIntergenic,sort(filteredData$FractionCoverage[which(filteredData$Region=="Intergenic")], decreasing = TRUE),  col = "blue", xlim=c(1,4800), ylim = c(0,1), xlab = "Index (arbitrary units)", ylab = "Fraction of sequence covered",lwd=0.5,cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5, yaxt="n", pch=19)
#legend(3000,0.96, c("Targets","Non-targets (Scaled)", "Intergenic (Scaled)"), #lty=c(1,1), lwd=c(4.5,4.5),col=c("violet","green","blue"), cex=2.0)
#eaxis(2, cex.lab=1.5, cex.axis=1.5)


FractionCoveragePlot<-ggplot(filteredData,aes(x=steps, y=FractionCoverage, colour = Region)) + geom_point()#, xlim=c(1,4800), ylim = c(0,1),  xlab = "Index (arbitrary units)", ylab = "Fraction of sequence covered")
FractionCoveragePlot
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/FractionCoveragePlot.pdf")
#dev.off()


fD2<-filteredData
fD2$DepthMean<-0
fD2$DepthSD<-0

fD2$DepthMean[which(fD2$Region=="Target")]<-mean(fD2$Depth[which(fD2$Region=="Target")])
fD2$DepthMean[which(fD2$Region=="Non-Target")]<-mean(fD2$Depth[which(fD2$Region=="Non-Target")])
fD2$DepthMean[which(fD2$Region=="Intergenic")]<-mean(fD2$Depth[which(fD2$Region=="Intergenic")])

fD2$DepthSD[which(fD2$Region=="Target")]<-sd(fD2$Depth[which(fD2$Region=="Target")])
fD2$DepthSD[which(fD2$Region=="Non-Target")]<-sd(fD2$Depth[which(fD2$Region=="Non-Target")])
fD2$DepthSD[which(fD2$Region=="Intergenic")]<-sd(fD2$Depth[which(fD2$Region=="Intergenic")])

Mean_Targets<-mean(fD2$Depth[which(fD2$Region=="Target")])
Mean_NonTargets<-mean(fD2$Depth[which(fD2$Region=="Non-Target")])
Mean_Intergenic<-mean(fD2$Depth[which(fD2$Region=="Intergenic")])



fD2$Depth[which(fD2$Depth<1e-2)]<-1e-2
#pdf(file="//Users/Vish/Downloads/Lorenzo/pool/FASTQ/violin_plot.pdf", width = 15, height = 15)
VIOLINplot <- ggplot(fD2, aes(x=Region, y=Depth, fill=Region)) + 
geom_violin()+
geom_jitter( color = "grey", size = 0.001)+ 
geom_point(aes(y = DepthMean), color = "black", size = 3, data = fD2) + 
geom_errorbar(aes(y = DepthMean, ymin = DepthMean, ymax = DepthMean+DepthSD), color = "black", width = 0.05, data = fD2)+
labs(x="Region", y = "Depth")+
theme(axis.title.y = element_text(face="bold", colour="black", size=20),
axis.text.x  = element_text(angle=90, vjust=0.5, size=16), axis.text.y  = element_text(size=16))+
  scale_y_log10(limits=c(1e-2,2000),expand = c(0, 0.1), breaks=c(.01,.1,1,10,100,1000),labels=c(.01,.1,1,10,100,1000))
VIOLINplot
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/VIOLINplot.pdf")
#dev.off()


fD2Target<-filteredData[which(filteredData$Region=="Target" & filteredData$Depth>=0.01 ),]
model_eqn=lm(Depth~log10(Length), fD2Target)
model_log=lm(log10(Depth)~Length, fD2Target)


 
#pdf(file="//Users/Vish/Downloads/Lorenzo/pool/FASTQ/depth_length.pdf", width = 15, height = 15)
DepthCapturePlot <- ggplot(fD2Target, aes(x=Length, y=Depth)) + 
geom_jitter( color = "blue",size = 1)+ 
geom_abline(intercept=round(model_log$coefficients[1],3),
                      slope=round(model_log$coefficients[2],3), color = "red", size = 2)+
  scale_y_log10(limits=c(1e-2,20000),expand = c(0, 0), breaks=c(0.1,1,10,100,1000,10000),labels=c(0.1,1,10,100,1000,10000))+
  scale_x_continuous(limits=c(0,5000),expand = c(0, -300), breaks=c(400,1000,2000,3000,4000,5000),labels=c(400,1000,2000,3000,4000,5000))+
labs(x="Length", y = "Depth")+
annotate("text", x = 2000, y = 1000, size=6, color="black",
           label = paste("y =", round(model_eqn$coefficients[2],2), "*log(x)","+",round(model_eqn$coefficients[1],2)))+
  annotate("text", x = 2400, y = 100, size=9, color="red",
           label = paste("R2 =", round(summary(model_log)$r.squared,3)))
DepthCapturePlot
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/DepthCapturePlot.pdf")
#dev.off()  

fC<-read.table("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/depthDHMVYVBCXX_1_GGCTAC_1.sorted.txt")
names(fC)<-c("chromosome","PositionLASSO","FrequencyLASSO")
# Read fasta file for MG1655 K-12 substr.
#targetsData<-read.table("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/targetsHMVYVBCXX_1_GGCTAC_1.out", header=FALSE)
f_gene<-targetsData
#names(f_gene)<-c("chromosome","Start","Stop", "Strand", "Gene", "Reads", "BasesCovered", "Length", "FractionCoverage")

library(IRanges)
iGene <- with(fC, IRanges(PositionLASSO, width=1, names=chromosome))
iDepth <- with(f_gene, IRanges(Start, Stop, names=Gene))
olaps <- findOverlaps(iGene, iDepth)
AlongGeneBases<-cbind(fC[queryHits(olaps),], f_gene[subjectHits(olaps),])
AlongGeneBases$PosIndex<-(AlongGeneBases$PositionLASSO-AlongGeneBases$Start)/AlongGeneBases$Length
filteredGeneCoverage <- subset(AlongGeneBases,Reads>=1000 & Reads<=1100, select=c(Gene, FrequencyLASSO, PosIndex, Reads))
#pdf(file="//Users/Vish/Downloads/Lorenzo/pool/FASTQ/PerBaseDepth_AlongGene.pdf", width = 15, height = 15)
GeneCovered <- ggplot(filteredGeneCoverage, aes(x=PosIndex, y=FrequencyLASSO, fill=Gene)) +
   geom_line( aes(colour=Gene), size = 0.9)+
  labs(x="Position Along Gene", y = "Per Base Depth")+
  scale_y_continuous(limits=c(0,300),expand = c(0, 0), breaks=c(0,50,100,200,300),labels=c(0,50,100,200,300))+
  guides(fill=FALSE)
GeneCovered
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/GeneCovered.pdf")
#dev.off()

missedTargets<-targetsData$Gene[which(targetsData$FractionCoverage<=0.0000001)]
headerEcoliFasta<-read.csv("/Users/Vish/Downloads/Lorenzo/metadata.bed", sep="\t", header = FALSE)
headerEcoliFasta$V4<-gsub('] ','',headerEcoliFasta$V4)
names(headerEcoliFasta)<-c("Chromosome","Accession","Gene","Description","Dirn","Start","Stop")
headerEcoliFasta$Length<-headerEcoliFasta$Stop-headerEcoliFasta$Start+1
missedList_dirn<-subset(headerEcoliFasta, toupper(Gene) %in% toupper(missedTargets), select=c(Accession, Description, Gene, Length))
write.csv(missedList_dirn, "/Users/Vish/Downloads/Lorenzo/paper/missedTargets_dirn.csv")

hitT<-length(which(targetsData$Reads>=1))
nohitT<-dim(targetsData)[1]-hitT
hitNT<-length(which(nontargetsData$Reads>=1))
nohitNT<-dim(nontargetsData)[1]-hitNT
LASSO_targets <-
matrix(c(hitT,nohitT,hitNT,nohitNT),
       nrow = 2,
       dimnames =
       list(c("True Positives", "False Positives"),
            c("Targets", "Non-Targets")))
fisher.test(LASSO_targets, conf.level = 0.99)$conf.int
FT<-fisher.test(LASSO_targets, conf.int = FALSE)
Sens<-hitT/(hitT+nohitT)
Spec<-nohitNT/(hitNT+nohitNT)

df<-data.frame()
WriteText<-ggplot(df)+
  scale_x_continuous(limits=c(0,5),expand = c(0, 0))+
  scale_y_continuous(limits=c(0,5),expand = c(0, 0))+
annotate("text", y = 4, x = 2, size=9, color="red",
          label = paste("p-value Fisher's Test =", FT$p.value))+
annotate("text", y = 3.5, x = 2, size=9, color="red",
          label = paste("Sensitivity =", round(Sens,3)))+
annotate("text", y = 3, x = 2, size=9, color="red",
          label = paste("Specificity =", round(Spec,3)))+
  annotate("text", y = 2.5, x = 2, size=9, color="red",
          label = paste("Mean Depth of Targets =", round(Mean_Targets,3)))+
annotate("text", y = 2, x = 2, size=9, color="red",
          label = paste("Mean Depth of Non-Targets =", round(Mean_NonTargets,3)))+
annotate("text", y = 1.5, x = 2, size=9, color="red",
          label = paste("Mean Depth of Intergenic Regions =", round(Mean_Intergenic,3)))
WriteText
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/WriteTextplot.pdf")

plot_grid(ROCplot, FractionCoveragePlot, VIOLINplot, DepthCapturePlot, GeneCovered, WriteText,  labels = "AUTO", ncol = 2)
ggsave("/Users/Vish/Downloads/Lorenzo/pool/FASTQ/AllPlots_GGCTAC.pdf", width = 30, height = 30)
```

