#---
#Author: Viswanadham Sridhara
#Date: 10 February 2019
#Title: Snakefile describing different aspects of the NGS pipeline used
#Project: Long Adapter Single Stranded Oligonucleotide (LASSO) Probes for Massively Multiplexed Cloning of Kilobase-Sized Genome Regions
#---
# To run the below pipeline, you need the reads and the fasta files.
# Below, you can find the information on how to get the reads from NCBI SRA portal and the fasta file (i.e., k12_profile.fasta)

# These SRRs are from the project description from NCBI portal
# https://www.ncbi.nlm.nih.gov/sra/SRX1981073[accn]
# Use sra-tools fasterq-dump to dump the SRRs used in the project

# The fasta used in this analyses is from
# https://www.ncbi.nlm.nih.gov/genome?term=NC_000913.3&cmd=DetailsSearch

all_ecoli_samples = ['SRR3956929', 'SRR5468269', 'SRR5468390', 'SRR5468391', 'SRR5468392', 'SRR5468393', 'SRR5468394']

ecoli_samples = ['SRR3956929']
#rule bowtie_index_build:
#    input:
#        "k12_profile.fasta"
#    shell:
#        "bowtie2-build {input} k12Index/k12_profile"


#rule bowtie_map:
#    input:
#        expand("{sample}.fastq", sample=ecoli_samples)
#    output:
#        expand("{sample}.sam", sample=ecoli_samples)
#    shell:
#        "bowtie2 -x k12Index/k12_profile -p 4 --very-sensitive --no-unal -U {input} -S {output} -k 1"


#rule samtools_view:
#    input:
#        expand("{sample}.sam", sample=ecoli_samples)
#    output:
#        expand("{sample}.bam", sample=ecoli_samples)
#    shell:
#        "samtools view -q 30 -bS {input} > {output}"


#rule samtools_sort:
#    input:
#        expand("{sample}.bam", sample=ecoli_samples)
#    output:
#        expand("{sample}.Sorted", sample=ecoli_samples)
#    shell:
#        "samtools sort {input} > {output}"

#rule bedtools_coverageBed:
#    input:
#        expand("{sample}.Sorted", sample=ecoli_samples)
#    output:
#        expand("{sample}.target", sample=ecoli_samples)
#    shell:
#        "coverageBed -b {input} -a regionIndex/targetk12chr.bed -wa > {output}"


#rule bedtools_coverageBed_depth:
#    input:
#        expand("{sample}.Sorted", sample=ecoli_samples)
#    output:
#        expand("{sample}.d_target", sample=ecoli_samples)
#    shell:
#        "coverageBed -b {input} -a regionIndex/targetk12chr.bed -d > {output}"

rule bedtools_run_once:
    input:
        "k12_profile.fasta"
    output:
        "targetsCoverageATGC.txt"
    shell:
        "bedtools nuc -fi {input} -bed regionIndex/targetk12chr.bed > {output}"
